frontend/src/app/api/api/events.service.ts

```diff
@@ -0,0 +1,304 @@
+                withCredentials: this.configuration.withCredentials,
+            }
+        );
+    }
+
+    /**
+     * Deletes a event
+     *
+     * @param id User type id to delete
+     */
+    public deleteEvent(id: number): Observable<{}> {
+        if (id === null || id === undefined) {
+            throw new Error('Required parameter id was null or undefined when calling deleteEvent.');
+        }
+
+        let headers = this.defaultHeaders;
+
+        // authentication (h2ms_auth) required
+        if (this.configuration.accessToken) {
+            let accessToken = typeof this.configuration.accessToken === 'function'
+                ? this.configuration.accessToken()
+                : this.configuration.accessToken;
+            headers = headers.set('Authorization', 'Bearer ' + accessToken);
+        }
+
+        // to determine the Accept header
+        let httpHeaderAccepts: string[] = [
+            'application/json'
+        ];
+        let httpHeaderAcceptSelected: string = this.configuration.selectHeaderAccept(httpHeaderAccepts);
+        if (httpHeaderAcceptSelected !== undefined) {
+            headers = headers.set('Accept', httpHeaderAcceptSelected);
+        }
+
+        // to determine the Content-Type header
+        let consumes: string[] = [];
+
+        return this.httpClient.delete<any>(`${this.basePath}/event/${encodeURIComponent(String(id))}`,
+            {
+                headers: headers,
+                withCredentials: this.configuration.withCredentials,
+            }
+        );
+    }
+
+    /**
+     * Get a list of all events
+     * Returns a list of all events
+     * @param perPage Results per page
+     * @param page Page number
+     * @param wardId Limit results to a ward
+     * @param hospitalId Limit results to a hospital
+     * @param subjectId Limit results to a particular user
+     * @param eventTypeId Limit results to an event type
+     */
+    public getEvent(perPage?: number, page?: number, wardId?: number, hospitalId?: number, subjectId?: number, eventTypeId?: number):
+    Observable<Array<Event>> {
+
+        let queryParameters = new HttpParams({encoder: new CustomHttpUrlEncodingCodec()});
+        if (perPage !== undefined) {
+            queryParameters = queryParameters.set('per_page', <any>perPage);
+        }
+        if (page !== undefined) {
+            queryParameters = queryParameters.set('page', <any>page);
+        }
+        if (wardId !== undefined) {
+            queryParameters = queryParameters.set('ward_id', <any>wardId);
+        }
+        if (hospitalId !== undefined) {
+            queryParameters = queryParameters.set('hospital_id', <any>hospitalId);
+        }
+        if (subjectId !== undefined) {
+            queryParameters = queryParameters.set('subject_id', <any>subjectId);
+        }
+        if (eventTypeId !== undefined) {
+            queryParameters = queryParameters.set('event_type_id', <any>eventTypeId);
+        }
```

Comment:
Duplication that seems to come up alot. Consider extracting a helper function.

queryParameters = Maps.setIfDefined(queryParameters, 'subject_id', <any>subjectId);